lib/models/local-user.js

```diff
@@ -0,0 +1,78 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = LocalUserModelFactory;
+
+LocalUserModelFactory.$provide = 'Models.LocalUser';
+LocalUserModelFactory.$inject = [
+    'Model'
+];
+
+var crypto = require('crypto');
+
+var hashConfig = {
+    // number of bytes in the Hash
+    hashBytes: 64,
+    // number of bytes in the salt
+    saltBytes: 64,
+    // number of iterations to get the final hash, longer means more secure,
+    // but also slower
+    iterations: 10000
+};
+
+
+function LocalUserModelFactory (Model) {
+    return Model.extend({
+        connection: 'mongo',
+        identity: 'localusers',
+        attributes: {
+            username: {
+                type: 'string',
+                required: true
+            },
+            password:{
+                type: 'string',
+                required: true,
+                minLength: 6,
+                maxLength: 50
+            },
+            role: {
+                type: 'string',
+                defaultsTo: 'ReadOnly'  // change to Constants
+            },
+            comparePassword: function (password) {
+                var combined = new Buffer(this.password, 'base64');
```

Comment:
The compare password maybe a common function not limit to user model ? For example, we need to hash BMC password, too. So we may extract it as a common function later. This is nice to have.